### ISSUE TITLE
[Bug] CollectionView items do not resize properly when item template is grid (Regression from 4.7 to 4.8) (Android & iOS)

### ISSUE THREAD

**Issue by lafritay** [xamarin/Xamarin.Forms#12175]
### Description / Steps to Reproduce

Scenario: In a `CollectionView` draw a list of items that is a grid with two columns such that:
1. The first column contains a `StackLayout` with two labels and the first label gets updated after the initial draw, possibly increasing it's length.
2. The second column contains any element.

See the attached project for an example. Launch the app. Click on Browse. Wait about 5 seconds for the view to update.

### Expected Behavior

When the update occurs, the items in the `CollectionView` resize so all of the content is shown. This works on 4.7.0.1351:

![image](https://user-images.githubusercontent.com/438824/93630515-25b2e080-f9b8-11ea-9d3e-9bb225632ad0.png)

### Actual Behavior

On 4.8.0.1451, the items change size but they get the wrong size which ends up with elements being cut off:

![image](https://user-images.githubusercontent.com/438824/93630655-53982500-f9b8-11ea-9fe8-5b174eab049e.png)

Notice how the item description line is cut off or hidden in a few of the rows. Unlike most other CollectionView item resize bugs, this happens on BOTH Android and iOS.

### Basic Information

- Version with issue: 4.8.0.1451
- Last known good version: 4.7.0.1351
- IDE: Visual Studio
- Platform Target Frameworks: 
  - iOS:  13
  - Android: 10.0 
- Android Support Library Version: 
- Nuget Packages: Minimal. Reproed with default Xamarin Project
- Affected Devices: Pixel 3 Q 10.0 Emulator is where screenshots come from

### Screenshots

See above

### Reproduction Link

[ItemSizingBugXF.zip](https://github.com/xamarin/Xamarin.Forms/files/5247459/ItemSizingBugXF.zip)

### Workaround

None that I'm aware of. This is blocking an upgrade to 4.8

**Comment by modplug** [comment]
Is there any workaround for this? 
We suddenly got reports on this from our customers and since we rely on certain 4.8 features I don't want to go through the hassle and downgrade Xamarin Forms while a proper fix is implemented

**Comment by lafritay** [comment]
@modplug I'm not aware of a workaround. I'm using the workarounds mentioned in this issue (https://github.com/xamarin/xamarin.forms/issues/10842) for iOS and those didn't seem to immediately fix the problem but I didn't go very deep there and I didn't try to do something similar for Android.

**Comment by hartez** [comment]
@lafritay I'm looking at this bug now, and hopefully we can find a solution soon. In the meantime, in your repro project you can work around the issue by changing the height of your grid row from "*" to "Auto".

**Comment by lafritay** [comment]
@hartez Thanks for the workaround. I was able to apply that to my app and get things working. While doing so, I did notice a layout issue even when `Auto` is used. I'm assuming they are related so I'm including the sample here. 

[App2.zip](https://github.com/xamarin/Xamarin.Forms/files/5402877/App2.zip)

Even though we're using `Auto` everywhere, notice that the description in rows 4 and 5 is still getting clipped:

![image](https://user-images.githubusercontent.com/438824/96470306-254a7700-11fc-11eb-889a-11e8921f723f.png)

Good luck with the fix!
